workloads/burn-in/install_hplbit_deps.sh

```diff
@@ -0,0 +1,239 @@
+#if [ ${BUILD_CUDA} && ${BUILD_HPCX} == 0 ]; then
+#	echo "Terminating script, no installations specified."
+# 	exit	
+#fi
+
+ofed_major=${ofed_major:-5.0}
+echo "HPC-X for Mellanox OFED $ofed_major will be downloaded from"
+
+case ${ofed_major} in
+   "5.0") HPCX_URL="http://www.mellanox.com/downloads/hpc/hpc-x/v2.7/hpcx-v2.7.0-gcc-MLNX_OFED_LINUX-5.0-1.0.0.0-ubuntu18.04-x86_64.tbz"; echo $HPCX_URL ;;
+   "4.7") HPCX_URL="http://www.mellanox.com/downloads/hpc/hpc-x/v2.7/hpcx-v2.7.0-gcc-MLNX_OFED_LINUX-4.7-1.0.0.1-ubuntu18.04-x86_64.tbz"; echo $HPCX_URL ;;
+   *) echo "Unable to find a matching MOFED version for HPCX, exiting." & exit 1 ;;
+esac
+
+#sets the directory where the burn in test is located, then sets where CUDA and HPC-X will downloaded and built
+export BASEDIR=${BASEDIR:-$(cd $(dirname 0) && pwd)}
+export BUILDDIR=${BUILDDIR:-/tmp/build.$$}
+#echo $BASEDIR
+#echo $BUILDDIR
+
+APPSDIR=${BASEDIR}/apps
+
+mkdir -p $APPSDIR
+mkdir -p $BUILDDIR
+
+
+#Variable definitions for CUDA and HPC-X Installs
+
+#CUDA Install Variables
+CUDA_VERSION=11.0.2
+CUDA_HOME=${APPSDIR}/cuda/${CUDA_VERSION}
+mkdir -p ${CUDA_HOME}
+#echo $CUDA_HOME
+
+#HPCX Install Variables
+
+#HPCX_URL="http://www.mellanox.com/downloads/hpc/hpc-x/v2.7/hpcx-v2.7.0-gcc-MLNX_OFED_LINUX-5.0-1.0.0.0-ubuntu18.04-x86_64.tbz"
+HPCX_VERSION=$(echo $HPCX_URL | grep -o '[^/]*$' | cut -f5-6 -d-)
+#echo $HPCX_VERSION
+HPCX_FN=$(basename ${HPCX_URL})
+HPCX_DIR=$(basename ${HPCX_FN} .tbz)
+
+#Build CUDA Toolkit and HPC-X if selected to do so (Yes by default)
+
+#Build CUDA Toolkit from Run File
+if [ $BUILD_CUDA == 1 ]; then
+    echo ""
+    echo "INFO: Installing CUDA Toolkit ${CUDA_VERSION}"
+    echo ""
+
+    CLOG=/tmp/cuda-installer.log
+    if [ -f ${CLOG} ]; then
+	    echo ""
+	    echo "INFO: The file ${CLOG} was found, trying to remove".
+	    rm -f ${CLOG}
+            if [ -f ${CLOG} ]; then
+		    echo ""
+		    echo "ERROR: Unable to remove ${CLOG} before installation of CUDA."
+		    echo "ERROR: Existence of this file without write permission will"
+
+echo "ERROR: cause the instalation of CUDA to trigger a segementation"
+		    echo "ERROR: fault.  Please have the file removed, and try the"
+		    echo "ERROR: Installation again."
+		    echo ""
+		    exit 1
+            fi
+	    echo "INFO: File ${CLOG} has been removed."
+	    echo ""
+    fi
+
+#Change to build directory where the CUDA run file is downloaded.  Remove the file if it is already present. Then run the installer to install the toolkit only. 
+    cd ${BUILDDIR}
+    CUDAPATH=http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/cuda_11.0.2_450.51.05_linux.run
+    CUDAFN=$(basename ${CUDAPATH})
+    rm -f ${CUDAFN}
+    wget http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/${CUDAFN}
+
+    sh ${CUDAFN} --installpath=${CUDA_HOME} --silent --toolkit
+
+    if [ $? -ne 0 ]; then
+	echo "ERROR: Install of CUDA failed.  Exiting"
```

Comment:
Spaces instead of tabs?

Reply:
I'm actually not sure why it is done this ways